README.md

```diff
@@ -2,17 +2,71 @@
+* parser
+
+jejichž vzájemná spolupráce je znázorněna na diagramu níže
+
+```mermaid
+flowchart LR
+    START             -->           operator(OPERATOR)
+    operator    -- urls.txt --->    downloader(DOWNLOADER)
+    downloader  -- *.html --->      parser(PARSER)
+    parser      -- *.JSON -->       operator
+    operator           -->          MATERIALIZATION
+```
+
+### Operator ###
+
+Operator vytváří frontu URL adres určených ke stažení pro _downloader_. Počáteční fronta je onigumu předaná pluginem. Plugin se také stará o doplňování nových URL adress do fronty. Nové URL adresy může plugin získávat například z naparsované podoby dat, která je vytvořena pomocí _parser_.
+
+Činnost _operatoru_ se skládá z:
+1. inicializace práce Oniguma na dané aplikaci
+2. kontroly stavu zpracovaných a nezpracovaných URL adres z výstupu
+_operatoru_, popř. dle zapsaných souborů
+3. načítání nezpracovaných URL adres ze strukturovaných dat stažené stránky
+4. zařazování nezpracovaných URL adres do fronty pro _downloader_
+5. mazání URL adres z fronty od _parseru_ po předání všech nových stránek
+v jejím obsahu do fronty pro _downloader_
+
+### Downloader ###
+
+Downloader stahuje obsah a metadata nezpracovaných URL adres.
+
+Činnost _downloader_ se skládá:
+
+1. načítání URL adres ke stažení z fronty
+2. stahování obsahu URL adres a případných metadat
+4. vytváření fronty stažených URL adres včetně jejich obsahu a metadat
+3. mazání zpracovaných URL adres z fronty
+
+### Parser ###
+
+Parsuje potřebná data ze staženého obsahu a metadat do strukturované podoby.
+
+Činnost _parseru_ se skládá:
+
+1. kontrolovat stav fronty se staženými URL adresami
+2. parsovat obsah a metadata stažených URL adres do strukturované podoby dat
+3. strukturovaná data ukládat do JSON souborů, jejichž jména jsou tvořena
+hashem z jejich URL adres
+4. aktivace pluginu na strukturovaná data
+5. mazat URL adresy z fronty stažených URL adres
 
-The application uses Spiders (Ruby modules) containing workflow and data-mining methods to get data from a website. A Spider’s public interface consists of workflows how to get all the wanted data from the server. Typically get a page, parse it and possibly get more pages using the parsed data.
+#### Aplikace (neboli pavouci) ####
 
-### Mechanism ###
+Ze strukturované podoby dat uložené v JSON souborech, vyscrapuje potřebné informace.
 
-Scraping starts by inserting the first workflow action on a queue. This action doesn’t take any input an thus doesn’t have to wait for any data to be parsed. Any action can queue another action that uses the data parsed from a download page. Downloading and parsing ends when there is no more actions to be launched.
+Charakter výstupních dat či informací je závislý na uživatelských potřebách a také podoby internetového obsahu. Je téměř nemožné vytvořit univerzálního pavouka splňujícího všechny požadavky z kombinace obou výše zmíněných. Proto je možné si nadefinovat vlastní plugin pro vlastní potřeby.
```

Comment:
```suggestion
Charakter výstupních dat či informací je závislý na uživatelských potřebách a také podoby internetového obsahu. Je téměř nemožné vytvořit univerzálního pavouka splňujícího všechny požadavky z kombinace obou výše zmíněných. Proto je nutné si nadefinovat vlastní plugin pro vlastní potřeby.
```

It is, in fact, necessary to write applications. Without them, Onigumo does nothing.